### PR TITLE
Checked spelling on Docs. Updated documentation for inbuilt functions and objects

### PR DESCRIPTION
**Pull request by W3NDO**

### CODE REVIEW

File: docs/guide/02-grammar-and-types.md [hunk 1]
```diff
@@ -2,26 +2,25 @@
...
 ## Basics
 
-- Swahili borrows its syntax from javascript, python and ruby. Like Javascript, Swahili is case sensitive. as such the declarations
+- Swahili borrows its syntax from Javascript, Python and Ruby. Like Javascript, Swahili is case sensitive. as such the declarations below are different variables.
 
 ```
-  wacha x = 5
-  wacha X = 3
+  wacha b = 5
+  wacha B = 3
```
Comment (W3NDO) [inline docs/guide/02-grammar-and-types.md:9]: Changed these for that stark difference between uppercase & lowercase

File: docs/guide/02-grammar-and-types.md [hunk 2]
```diff
@@ -79,15 +79,15 @@ x = 5 // this will not raise an error
...
-  - Kamusi : Works like a dictionary by storing items as **key/value pairs**
+- There are 8 Data structures and Types in Swahili.
+  1. Boolean: `kweli` that represents true and `uwongo` that represents false
+  2. null: `tupu` denotes a null value
+  3. Number: Could be both Integers and floats. For example `42` or `3.14159`
+  4. Strings: A sequence of characters that represents a text value. For example `Habari`
+  5. Lists: `Orodha` an ordered collection of items. Can hold multiple data types: for example `[1,2,3]` or `[1, "orodha", kweli]`
+  6. Regular Expressions (RegEx) : These are patterns used to match character combinations in strings. 
+  7. Date : Used to represent a date object.
+  8. Kamusi : Works like a dictionary by storing items as **key/value pairs**
```
Comment (malcolmkiano) [inline docs/guide/02-grammar-and-types.md:90]: Naming inconsistency (all the rest are in English)

File: docs/ref/built-in-functions.md [hunk 3]
```diff
@@ -16,8 +16,21 @@
 
 ## Type methods
```
Comment (malcolmkiano) [inline docs/ref/built-in-functions.md:17]: We may want to indicate the syntax for calling these type methods

**Review by malcolmkiano — COMMENTED** [review]
Maybe a little segment under each of the type methods showing an example of the call and the output

File: docs/ref/built-in-functions.md (hunk 3 above)
Comment (W3NDO) [reply docs/ref/built-in-functions.md:17]: Added that in the current edit I'm working on. Decided to add them in the Grammar and Types documentation instead

File: docs/guide/02-grammar-and-types.md (hunk 2 above)
Comment (W3NDO) [reply docs/guide/02-grammar-and-types.md]: Do we call them Dictionaries or Objects?

**Review by malcolmkiano — APPROVED** [review]

**Pull request merged**